common/lib/dependabot/config/update_config.rb

```diff
@@ -1,32 +1,25 @@
 # frozen_string_literal: true
 
+require "dependabot/config/ignore_condition"
+
 module Dependabot
   module Config
     # Configuration for a single ecosystem
     class UpdateConfig
-      attr_reader :commit_message_options
-
+      attr_reader :commit_message_options, :ignore_conditions
       def initialize(ignore_conditions: nil, commit_message_options: nil)
         @ignore_conditions = ignore_conditions || []
         @commit_message_options = commit_message_options
       end
 
-      def ignored_versions_for(dep)
+      def ignored_versions_for(dependency)
         @ignore_conditions.
-          select { |ic| ic.dependency_name == dep.name }. # FIXME: wildcard support
-          map(&:versions).
+          select { |ic| ic.dependency_name == dependency.name }. # FIXME: wildcard support
+          map { |ic| ic.ignored_versions(dependency) }.
```

Comment:
Concern: as I as alluding to in https://github.com/dependabot/dependabot-core/pull/3536/files#r617772124 , this isn't going to behave when multiple conditions match the target dependency...

Reply:
Not sure I follow, do we have a test case for this that will fail currently?

I was looking at moving the dependency name checking into the IgnoreCondition class but would mean returning an empty `ignored_versions` if the dependency name doesn't match the `ic.dependency_name` which seems confusing to me but unsure if I'm missing something else?

Reply:
> this isn't going to behave when multiple conditions match the target dependency

Is this case tested here?: https://github.com/dependabot/dependabot-core/pull/3513/files#diff-1bee1e376554d0d150b3a22f7b1ca87ac8d9de442cebf942c2f0e3f7fcc50c04R56